apps/client/src/liquidityVenues/midas/index.ts

```diff
@@ -0,0 +1,329 @@
+        tokenOutDecimals,
+      ] = await Promise.all([
+        this.getRedemptionVaultMinAmount(vault, encoder),
+        this.getRedemptionVaultInstantFee(vault, encoder),
+        this.getRedemptionVaultInstantDailyLimit(vault, encoder),
+        this.getRedemptionVaultStableCoinRate(vault, encoder),
+        this.getRedemptionVaultWaivedFeeRestriction(vault, encoder),
+        this.getRedemptionVaultDailyLimits(vault, encoder),
+        this.getRedemptionVaultmTokenDataFeed(vault, encoder),
+        this.getRedemptionVaultMTokenOutConfig(vault, tokenOut, encoder),
+        readContract(encoder.client, {
+          address: tokenOut,
+          abi: erc20Abi,
+          functionName: "decimals",
+          args: [],
+        }),
+      ]);
+
+      const [mTokenRate, tokenOutRate] = await Promise.all([
+        this.getMidasRate(mTokenDataFeed, encoder),
+        this.getMidasRate(tokenOutConfig[0], encoder),
+      ]);
+
+      return {
+        amountMTokenIn: seizedCollateral,
+        tokenOutConfig: {
+          dataFeed: tokenOutConfig[0],
+          fee: tokenOutConfig[1],
+          allowance: tokenOutConfig[2],
+          stable: tokenOutConfig[3],
+        },
+        tokenOutDecimals: BigInt(tokenOutDecimals),
+        minAmount,
+        instantFee,
+        instantDailyLimit,
+        STABLECOIN_RATE,
+        waivedFeeRestriction,
+        dailyLimits,
+        mTokenRate,
+        tokenOutRate,
+      };
+    } catch (e) {
+      console.error(e);
+      return undefined;
+    }
+  }
+
+  async getRedemptionVaultMinAmount(vault: Address, encoder: ExecutorEncoder) {
+    return readContract(encoder.client, {
+      address: vault,
+      abi: redemptionVaultAbi,
+      functionName: "minAmount",
+      args: [],
+    });
+  }
+
+  async getRedemptionVaultInstantFee(vault: Address, encoder: ExecutorEncoder) {
+    return readContract(encoder.client, {
+      address: vault,
+      abi: redemptionVaultAbi,
+      functionName: "instantFee",
+      args: [],
+    });
+  }
+
+  async getRedemptionVaultInstantDailyLimit(vault: Address, encoder: ExecutorEncoder) {
+    return readContract(encoder.client, {
+      address: vault,
+      abi: redemptionVaultAbi,
+      functionName: "instantDailyLimit",
+      args: [],
+    });
+  }
+
+  async getRedemptionVaultStableCoinRate(vault: Address, encoder: ExecutorEncoder) {
+    return readContract(encoder.client, {
+      address: vault,
+      abi: redemptionVaultAbi,
+      functionName: "STABLECOIN_RATE",
+      args: [],
+    });
+  }
+
+  async getRedemptionVaultWaivedFeeRestriction(vault: Address, encoder: ExecutorEncoder) {
+    return readContract(encoder.client, {
+      address: vault,
+      abi: redemptionVaultAbi,
+      functionName: "waivedFeeRestriction",
+      args: [encoder.address],
+    });
+  }
+
+  async getRedemptionVaultDailyLimits(vault: Address, encoder: ExecutorEncoder) {
+    const currentDayNumber = Math.round(Date.now() / 1000 / (60 * 60 * 24));
```

Comment:
Is rounding appropriate or should it be floored? I'm not familiar with contracts.